### DISCUSSION TITLE
dirtyFields are not updated when I change radio inputs with initial values

### DISCUSSION THREAD

**Discussion by brilovichgal** [react-hook-form/react-hook-form#11715] [Q&A]
Hi, 
I think there is a bug in react hook form when dealing with dirtyFields with input of type "radio" with initial value.
When changing the value of the input to some other value ("alon" for example), the formState.dirtyFields keeps being empty, and also the formState.isDirty is false.
This happens in the specific situation when i have set initial value to the radio, and try to change to other value.
Like in this code example:
```export default function App() {
  const form = useForm({
    defaultValues: {
      hello: "world"
    },
    mode: "onChange"
  });
  const { register, formState } = form;

  return (
    <form>
      <label>
        <div>
          <input type="radio" value="dor" {...register("hello")} />
          dor
        </div>
      </label>
      <label>
        <div>
          <input type="radio" value="world" {...register("hello")} />
          world
        </div>
      </label>
      <label>
        <div>
          <input type="radio" value="alon" {...register("hello")} />
          alon
        </div>
      </label>
      <pre>{JSON.stringify(form.watch(), null, 2)}</pre>
    </form>
  );
}
`

**Comment by sujinleeme** [comment]
I haven't found any problem with dirtyFields in your code snippet.

When a user clicks the alon radio button:
![image](https://github.com/react-hook-form/react-hook-form/assets/6011401/f9da6d39-6038-42fe-945a-59d5ee8df371)

When a user clicks the world radio button:
![image](https://github.com/react-hook-form/react-hook-form/assets/6011401/13cf830f-31e7-4e95-99ad-38926642e87b)
